utils/__tests__/build-validate-quickstart-artifact.test.js

```diff
@@ -0,0 +1,163 @@
+import * as fs from 'fs';
+import Quickstart from '../lib/Quickstart';
+import DataSource from '../lib/DataSource';
+import Alert from '../lib/Alert';
+import Dashboard from '../lib/Dashboard';
+
+import {
+  getArtifactComponents,
+  getDataSourceIds,
+  validateArtifact,
+} from '../build-validate-quickstart-artifact';
+
+jest.mock('../lib/Quickstart');
+jest.mock('../lib/DataSource');
+jest.mock('../lib/Alert');
+jest.mock('../lib/Dashboard');
+jest.mock('fs');
+
+describe('built-validate-quickstart-artifact', () => {
+  beforeEach(() => {
+    // disable normal logging used as feedback while script runs
+    jest.spyOn(console, 'log').mockImplementation(() => {});
+  });
+
+  describe('build', () => {
+    it('should find all of the components', () => {
+      Quickstart.getAll = jest
+        .fn()
+        .mockReturnValueOnce([
+          { config: 'test-quickstart-1' },
+          { config: 'test-quickstart-2' },
+        ]);
+
+      DataSource.getAll = jest
+        .fn()
+        .mockReturnValueOnce([{ config: 'test-dataSource-1' }]);
+
+      Alert.getAll = jest.fn().mockReturnValueOnce([]);
```

Comment:
we might want to test each of these with non-empty values here since we had problems with the getAll returning nothing before!